src/common/contexts/provider.tsx

```diff
@@ -49,44 +74,118 @@ declare global {
 
 interface ProviderContextProviderProps {
   children: React.ReactNode;
+  networks: ChainInfoObject[];
+  defaultChainId: ChainId;
 }
 
-export const ProviderContextProvider: FunctionComponent<ProviderContextProviderProps> = ({ children }) => {
+export const ProviderContextProvider: FunctionComponent<ProviderContextProviderProps> = ({
+  children,
+  networks: supportedChainInfoObjects,
+  defaultChainId,
+}) => {
   const [providerType, setProviderType] = useState<SIGNER_TYPE>(SIGNER_TYPE.IDENTITY);
-  const [provider, setProvider] = useState<providers.Provider | Signer>(getProvider);
-  const [account, setAccount] = useState<string>();
+  const [isConnected, setIsConnected] = useState<boolean>();
+  const [currentChainId, setCurrentChainId] = useState<ChainId | undefined>(defaultChainId);
 
-  const initializeMetaMaskSigner = async () => {
-    const { ethereum, web3 } = window;
+  const isSupportedNetwork = (chainId: ChainId) =>
+    supportedChainInfoObjects.findIndex((chainInfoObj) => chainInfoObj.chainId === chainId) > -1;
+
+  const defaultProvider = isSupportedNetwork(defaultChainId) ? createProvider(defaultChainId) : undefined;
+  const [providerOrSigner, setProviderOrSigner] = useState<providers.Provider | Signer | undefined>(defaultProvider);
+
+  const updateProviderOrSigner = async (newProviderOrSigner: typeof providerOrSigner) => {
+    try {
+      if (!Signer.isSigner(newProviderOrSigner)) {
```

Comment:
oh wow did you read the source code in order to discover the static methods for the class Signer?

Reply:
ohh... I think i saw it in the docs: https://docs.ethers.io/v5/api/signer/#Signer-isSigner

Reply:
ahhh okok, i didnt see that 👌